src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/UpdateAzureMLWebService.cs

```diff
@@ -0,0 +1,195 @@
+using Microsoft.WindowsAzure.Commands.Common;
+
+namespace Microsoft.Azure.Commands.MachineLearning
+{
+    [Cmdlet(
+        VerbsData.Update, 
+        WebServicesCmdletBase.CommandletSuffix, 
+        SupportsShouldProcess = true)]
+    [OutputType(typeof(WebService))]
+    public class UpdateAzureMLWebService : WebServicesCmdletBase
+    {
+        protected const string UpdateFromArgumentsParameterSet = 
+            "Update specific properties of the .";
+        protected const string UpdateFromObjectParameterSet = 
+            "Create a new Azure ML webservice from a WebService instance definition.";
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the resource group for the Azure ML web service.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the web service.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New title for the web service.")]
+        public string Title { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New description for the web service.")]
+        public string Description { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Mark the service as readonly.")]
+        public SwitchParameter IsReadOnly { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New access keys for the web service.")]
+        public WebServiceKeys Keys { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New access key for the storage account associated with the web service. This allows for key rotation if needed.")]
+        public string StorageAccountKey { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New diagnostics settings for the web service.")]
+        public DiagnosticsConfiguration Diagnostics { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New realtime endpoint runtime settings for the web service.")]
+        public RealtimeConfiguration RealtimeConfiguration { get; set; }
```

Comment:
If this is an enum, this shoudl be fine.  If not, how does the user create this?